### ISSUE TITLE
[V4] Importing of Fullcalendar Resource

### ISSUE THREAD

**Issue by ganySA** [fullcalendar/fullcalendar#4579]
Hi

In version 4.0 when i try import the resource core in angular using:

import '@fullcalendar/resource-common';

It seems to cause havoc, what is the correct ES6 way of importing the resource-common?

All my imports in my angular component as follows:

import { Calendar } from '@fullcalendar/core';
import momentPlugin from '@fullcalendar/moment';
import {toMoment } from '@fullcalendar/moment';
import dayGridPlugin from '@fullcalendar/daygrid';
import timeGridPlugin from '@fullcalendar/timegrid';
import listGridPlugin from '@fullcalendar/list';
import '@fullcalendar/resource-common';
import resourceTimeGridPlugin from '@fullcalendar/resource-timegrid';
import resourceDayGridPlugin from '@fullcalendar/resource-daygrid';
import momentTimezonePlugin from '@fullcalendar/moment-timezone';

It seems to break the entire calendar - multiple typescript compilation errors etc.

```
ERROR in node_modules/@fullcalendar/core/main.d.ts(2724,14): error TS2484: Export declaration conflicts with exported declaration of 'OptionsInput'.
node_modules/@fullcalendar/core/main.d.ts(2725,26): error TS2484: Export declaration conflicts with exported declaration of 'EventDef'.
node_modules/@fullcalendar/core/main.d.ts(2751,14): error TS2484: Export declaration conflicts with exported declaration of 'Calendar'.
node_modules/@fullcalendar/core/main.d.ts(2758,24): error TS2484: Export declaration conflicts with exported declaration of 'DateSpanApi'.
node_modules/@fullcalendar/core/main.d.ts(2758,37): error TS2484: Export declaration conflicts with exported declaration of 'DatePointApi'.
node_modules/@fullcalendar/core/main.d.ts(2778,35): error TS2484: Export declaration conflicts with exported declaration of 'CalendarState'.
node_modules/@fullcalendar/core/main.d.ts(2787,14): error TS2484: Export declaration conflicts with exported declaration of 'EventMutation'.
node_modules/@fullcalendar/core/main.d.ts(2789,14): error TS2484: Export declaration conflicts with exported declaration of 'EventApi'.
```
What is the correct way to import this?

**Comment by acerix** [comment]
This page has some info:

https://fullcalendar.io/docs/initialize-es6

And there is a demo project here:

https://github.com/fullcalendar/scheduler-typescript-example

**Comment by ganySA** [comment]
Hi

when i use the imports as per the example in my application and as per the demo it works fine.

```
import { Calendar } from '@fullcalendar/core';
import interactionPlugin from '@fullcalendar/interaction';
import dayGridPlugin from '@fullcalendar/daygrid';
import listPlugin from '@fullcalendar/list';
import timeGridPlugin from '@fullcalendar/timegrid';
import resourceTimelinePlugin from '@fullcalendar/list';
import momentPlugin from '@fullcalendar/moment';
import {toMoment } from '@fullcalendar/moment';
```
The minute i add the following to the list above - it starts throwing typescript panics as per my first post.:

```
import resourceTimeGridPlugin from '@fullcalendar/resource-timegrid';
import resourceDayGridPlugin from '@fullcalendar/resource-daygrid';
```

**Comment by acerix** [comment]
Ahh, I didn't read close enough.   There is no `resource-timegrid`, you would just use `timegrid` and set the resources to show resources in the view, same for `daygrid`.

**Comment by ganySA** [comment]
I’m pretty confused... so what import do I use? And how do I set the plugins on the calendar? I think the documentatio states resourceGridplugin must be added - if I don’t import it how can I reference it on the plugins properly for the calendar - sorry I’m a little confused.

If I don’t add resource plugin - I think the resource property in the calendar is also invalid...

**Comment by ganySA** [comment]
The documentation here specifies it must be imported and added as a plugin...

https://fullcalendar.io/docs/resource-daygrid-view

**Comment by acerix** [comment]
Oh sorry, I missed that page and didn't realise it had changed.

But, I added this code to the example project:

```
import resourceTimeGridPlugin from '@fullcalendar/resource-timegrid';
import resourceDayGridPlugin from '@fullcalendar/resource-daygrid';
```
At first I got "Cannot find module", but it works find after installing them like this:

`npm install --save @fullcalendar/resource-daygrid @fullcalendar/resource-timegrid`

So it sounds like you are doing it correctly, maybe you need to update TypeScript or clear some old files that are cached.

**Comment by ganySA** [comment]
I did that on the sample project too. It did compile - I just cannot find what’s different I deleted node_modules on mine and I’m on the very latest typescript.

**Comment by ganySA** [comment]
Solved - Typescript was at 3.1.1 and this needs 3.2.4+

**Comment by luiszarg96** [comment]
how should i upgrade my typescript?
